### PR TITLE
doc: Cleanup dead links in the documentation.

### PR DESCRIPTION
**Pull request by kartben**
I had a go at fixing the most obvious (and fixable) dead links in the documentation with the help of Sphinx's link checker.

### CODE REVIEW

**Review by greg-leach — DISMISSED** [review]

File: boards/arm/tdk_robokit1/doc/index.rst [hunk 1]
```diff
@@ -107,7 +107,7 @@ features:
      - N/A
 
 The default configuration can be found in the Kconfig
-:zephyr_file:`boards/arm/tdk_robokit1/tdk_robotkit1_defconfig`.
+:zephyr_file:`boards\arm\tdk_robokit1\tdk_robokit1_defconfig`.
```
[the remark this answers is not in the document]
Comment (kartben) [reply boards/arm/tdk_robokit1/doc/index.rst:110]: typo! the actual fix is robo**t**kit --> robokit

**Review by sylvioalves — DISMISSED** [review]

**Review by thedjnK — APPROVED** [review]
Approving for auxdisplay

**Review by thedjnK — CHANGES_REQUESTED** [review]
Actually NACK for auxdisplay, seems that link to display is not working: https://builds.zephyrproject.io/zephyr/pr/58913/docs/hardware/peripherals/auxdisplay.html

**Review by nashif — APPROVED** [review]

**Review by thedjnK — APPROVED** [review]

**Comment by kartben** [comment]
> Actually NACK for auxdisplay, seems that link to display is not working: https://builds.zephyrproject.io/zephyr/pr/58913/docs/hardware/peripherals/auxdisplay.html

thanks for paying attention! Fixed.

**Review by nordicjm — APPROVED** [review]
OK for sysbuild and mcumgr

**Review by gmarull — APPROVED** [review]

**Pull request merged**